### DISCUSSION TITLE
Upgrade code base from 2.0.1 to latest

### DISCUSSION THREAD

**Discussion by ghost** [oqtane/oqtane.framework#2819] [Q&A]
Hi! We have a code base with version 2.0.1 downloaded from the release section. Then we have a lot of modules that are created inside the Oqtane projects. In the future, we will shift them and use them as an external module. We also have some changes to oqtane code itself as well.

What are the options for upgrading the project and database to the latest version? I've tried updating via uploading the package in the system update section but that didn't work.

**Comment by sbwalker** [comment]
@farhanalie if you modified the core framework code then unfortunately your only option is to identify your custom changes and integrate them into the latest version manually.  This is why it is highly recommended to develop your custom modules and themes independently from the framework.
